pact/http/provider/src/test/kotlin/provider/ContractTestWithPactBroker.kt

```diff
@@ -0,0 +1,121 @@
+package provider
+
+import au.com.dius.pact.provider.junit5.HttpTestTarget
+import au.com.dius.pact.provider.junit5.PactVerificationContext
+import au.com.dius.pact.provider.junit5.PactVerificationInvocationContextProvider
+import au.com.dius.pact.provider.junitsupport.Provider
+import au.com.dius.pact.provider.junitsupport.State
+import au.com.dius.pact.provider.junitsupport.VerificationReports
+import au.com.dius.pact.provider.junitsupport.loader.PactBroker
+import au.com.dius.pact.provider.junitsupport.loader.PactBrokerAuth
+import au.com.dius.pact.provider.junitsupport.loader.PactFolder
+import au.com.dius.pact.provider.junitsupport.loader.VersionSelector
+import io.mockk.clearAllMocks
+import io.mockk.every
+import io.mockk.mockk
+import org.apache.hc.core5.http.HttpRequest
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS
+import org.junit.jupiter.api.TestTemplate
+import org.junit.jupiter.api.condition.DisabledIfEnvironmentVariable
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT
+import org.springframework.boot.test.web.server.LocalServerPort
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.ComponentScan
+import org.springframework.http.HttpHeaders.AUTHORIZATION
+import org.springframework.http.HttpHeaders.encodeBasicAuth
+import provider.books.Book
+import provider.books.BookRecord
+import provider.books.Library
+import java.util.UUID
+import kotlin.text.Charsets.UTF_8
+
+// Example of provider-side contract test using the pact broker.
+// Verification results will be uploaded every time the test is run,
+// based on the value of the `pact.verifier.publishResults` property
+
+@TestInstance(PER_CLASS) // PACT needs test instance per class ... for some reason ...
+@Provider("provider") // load all contracts for the provider named "provider"
+@PactBroker( // configure the connection to the pact broker and which pacts to use
+    url = "http://localhost:9292",
+    authentication = PactBrokerAuth(
+        username = "pact-user",
+        password = "123"
```

Comment:
Not a change request, but a question for future optimization:
Did you find out if and how credentials could be externalized? e.g. System Properties?

Reply:
The free version of the broker (there's an enterprise tier at pactflow) only supports basic auth for read-ony/read-write access. For anything more sophisticated you'll have to pay. So externalization is doable via the usual means like environment variables etc.